DeleteNodeCommand test class

CharlotteXiaoYeDing opened the pull request:
There is code duplication between removeNode(), insertNode() and layout(). 
aNeedsLayout is set to be true twice.

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+import ca.mcgill.cs.stg.jetuml.graph.ClassNode;
+import ca.mcgill.cs.stg.jetuml.graph.Graph;
+import ca.mcgill.cs.stg.jetuml.graph.Node;
+
```
Please activate Checkstyle and use the style file in `/style/Style.xml`

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+    private Node aNode;
+    private ArrayList<Node> aRootNodes;
+    private DeleteNodeCommand aDeleteNodeCommand;
+    private boolean hasExecuted = false;
```
`hasExecuted` should be named with the usual convention for fields. More importantly, I don't get how this field is supposed to be set as a result of the command being executed.

CharlotteXiaoYeDing replied:
This is attribute is to make sure that we execute() the command before undo() it.

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+    private Field aNeedsLayout;
+    private Field aNodesToBeRemoved;
+    private Node aNode;
+    private ArrayList<Node> aRootNodes;
```
`aRootNodes` ends up only being an alias for the object inside the graph. I would remove the field and access the value locally when needed.

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+    private boolean hasExecuted = false;
+    
+    @Before
+    public void setUp() throws Exception
```
catch all the relevant exception types in a catch clause that calls `fail()`, it's cleaner.

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+    public void testExecute() {
+        aDeleteNodeCommand.execute();
+        hasExecuted = true;
+        try {
```
You can encapsulate the exception handling in a method. See how it's done in `TestPropertyChangeTracker.getPropertyName`

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,79 @@
+    
+    @Test
+    public void testUndo() {
+        if (!hasExecuted)
```
Tests should be independent

CharlotteXiaoYeDing commented:
I'll do the same changes for the other tests

prmr commented on .checkstyle:
```diff
@@ -1,13 +1,13 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Please remove this file from the commit

prmr commented on test/ca/mcgill/cs/stg/jetuml/commands/TestDeleteNodeCommand.java:
```diff
@@ -0,0 +1,78 @@
+    public void testUndo() 
+    {
+        aDeleteNodeCommand.execute();
+        int numOfNodes = aGraph.getRootNodes().size();
```
Just to be explicit you can put in `assertEquals(0, aGraph.getRootNodes().size());`

prmr commented:
You can remove the statement `aGraph.layout();` in both `DeleteNodeCommand.execute()` and `DeleteNodeCommand.undo()`, because they are indeed redundant.

CharlotteXiaoYeDing commented:
Hi Martin,

I’ve deleted .checkstyle and removed layout() in DeleteNodeCommand. Do I need to remove layout() in other commands?


> On May 25, 2016, at 15:29, Martin Robillard <notifications@github.com> wrote:
> 
> You can remove the statement aGraph.layout(); in both DeleteNodeCommand.execute() and DeleteNodeCommand.undo(), because they are indeed redundant.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/prmr/JetUML/pull/168#issuecomment-221681102>
